tests/test_core.py

```diff
@@ -199,6 +199,26 @@ def test_generate_from_jams(atol=1e-5, rtol=1e-8):
                          pitch_shift=('uniform', -1, 1),
                          time_stretch=('uniform', 0.8, 1.2))
 
+
+        def _validate_soundscape_and_event_audio(orig_wav_file, gen_wav_file):
+            # validate audio
+            orig_wav, sr = soundfile.read(orig_wav_file.name)
+            gen_wav, sr = soundfile.read(gen_wav_file.name)
+            assert np.allclose(gen_wav, orig_wav, atol=atol, rtol=rtol)
+
+            # validate event audio is trimmed and sums to trimmed soundscape
+            events_folder = gen_wav_file.name[:-4] + '_events'
+            gen_event_files = [
+                os.path.join(events_folder, x)
+                for x in sorted(os.listdir(events_folder))
+            ]
+            gen_audio = [soundfile.read(x)[0] for x in gen_event_files]
+
+            assert np.allclose(gen_wav, sum(gen_audio), atol=atol, rtol=rtol)
+
+            # clean up events folder
+            shutil.rmtree(events_folder, ignore_errors=True)
```

Comment:
Do we need this call to `rmtree`, if everything is happening in tempfiles under a context that will close them?

Reply:
I found that we did need it as sometimes my OS would allocate the same tempfile name, which keeps around the directory of the `_events` folder corresponding to that audio file. So you'd end up with event audio from the previous iteration of the loop. Very strange, but this was put in explicitly to fix that. I'll add a more detailed comment.